p2p/src/sync/peer.rs

```diff
@@ -144,6 +150,10 @@ where
                 block_to_send_to_peer = async { self.blocks_queue.pop_front().expect("The block queue is empty") }, if !self.blocks_queue.is_empty() => {
                     self.send_block(block_to_send_to_peer).await?;
                 }
+
+                _ = stalling_interval.tick(), if self.last_activity.is_some() => {
```

Comment:
By default, tokio interval uses `MissedTickBehavior::Burst`: https://docs.rs/tokio/latest/tokio/time/fn.interval.html
And so once `last_activity` is some, `stalling_interval` may fire a lot of missed ticks which is not really needed.